lib/stripe/stripe_client.rb

```diff
@@ -7,21 +7,29 @@ module Stripe
   # recover both a resource a call returns as well as a response object that
   # contains information on the HTTP call.
   class StripeClient
+    include Stripe::ClientAPIOperations
+
     # A set of all known thread contexts across all threads and a mutex to
     # synchronize global access to them.
     @thread_contexts_with_connection_managers = []
     @thread_contexts_with_connection_managers_mutex = Mutex.new
     @last_connection_manager_gc = Util.monotonic_time
 
-    # Initializes a new `StripeClient`.
-    #
-    # Takes a connection manager object for backwards compatibility only, and
-    # that use is DEPRECATED.
-    def initialize(_connection_manager = nil)
+    # Initializes a new StripeClient
+    def initialize(config_overides = {})
       @system_profiler = SystemProfiler.new
       @last_request_metrics = nil
+      @config_overides = config_overides
     end
 
+    # Always base config off the global Stripe configuration to ensure the
+    # client picks up and changes to the config.
```

Comment:
Typo - should be `picks up any changes`. Will update with any other feedback to prevent rebase spam.

Reply:
Sounds good.